desc/objectives/_generic.py

```diff
@@ -11,12 +12,160 @@
 from desc.compute.utils import _parse_parameterization, get_profiles, get_transforms
 from desc.grid import QuadratureGrid
 from desc.optimizable import OptimizableCollection
-from desc.utils import errorif, parse_argname_change
+from desc.utils import errorif, jaxify, parse_argname_change
 
 from .linear_objectives import _FixedObjective
 from .objective_funs import _Objective, collect_docs
 
 
+class ExternalObjective(_Objective, ABC):
+    """Wrap an external code.
+
+    Similar to ``ObjectiveFromUser``, except derivatives of the objective function are
+    computed with finite differences instead of AD. The function does not need not be
+    JAX transformable.
+
+    The user supplied function must take an Equilibrium or a list of Equilibria as its
+    only positional argument, but can take additional keyword arguments.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
+    fun : callable
+        External objective function. It must take an Equilibrium or list of Equilibria
+        as its only positional argument, but can take additional kewyord arguments.
+        It does not need to be JAX transformable.
+    dim_f : int
+        Dimension of the output of ``fun``.
+    vectorized : bool
+        Set to False if ``fun`` takes a single Equilibrium as its positional argument.
+        Set to True if ``fun`` instead takes a list of Equilibria.
+    abs_step : float, optional
+        Absolute finite difference step size. Default = 1e-4.
+        Total step size is ``abs_step + rel_step * mean(abs(x))``.
+    rel_step : float, optional
+        Relative finite difference step size. Default = 0.
+        Total step size is ``abs_step + rel_step * mean(abs(x))``.
+    kwargs : any, optional
+        Keyword arguments that are passed as inputs to ``fun``.
+
+    """
+
+    __doc__ = __doc__.rstrip() + collect_docs(
+        target_default="``target=0``.", bounds_default="``target=0``."
+    )
+
+    _units = "(Unknown)"
+    _print_value_fmt = "External objective value: "
+    _static_attrs = ["_fun_wrapped", "_kwargs"]
+
+    def __init__(
+        self,
+        eq,
+        *,
+        fun,
+        dim_f,
+        vectorized,
+        abs_step=1e-4,
+        rel_step=0,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=False,
+        normalize_target=False,
+        loss_function=None,
+        name="external",
+        **kwargs,
```

Comment:
This assumes that any kwargs are to be passed to the user function, which might cause issues down the road if we need kwargs for anything else (like handling an argument renaming). Would be cleaner if instead the kwargs to the user function were just passed as a dict like
```
fun_kwargs={"a": True, "b": 1.2}